Samples/EmbeddedAuthWithSDKs/EmbeddedAuthUITests/EmailMFALoginScenarioTests.swift

```diff
@@ -20,54 +20,76 @@ final class EmailMFALoginScenarioTests: ScenarioTestCase {
         try scenario.createUser(groups: [.mfa])
     }
     
-    func testLoginWithEmail() throws {
+    func test_Login_With_MFA_Email() throws {
         let credentials = try XCTUnwrap(scenario.credentials)
         let signInPage = SignInFormPage(app: app)
         signInPage.signIn(username: credentials.username, password: credentials.password)
         
-        let factorsPage = FactorsEnrollmentPage(app: app)
-        XCTAssertTrue(factorsPage.emailLabel.waitForExistence(timeout: .regular))
-        XCTAssertTrue(factorsPage.chooseButton.exists)
+        test("THEN she is presented with an option to select Email to verify") {
+            let factorsPage = FactorsEnrollmentPage(app: app)
+            XCTAssertTrue(factorsPage.emailLabel.waitForExistence(timeout: .regular))
+            XCTAssertTrue(factorsPage.chooseButton.exists)
+            
+            test("WHEN She selects Email from the list") {
+                factorsPage.emailLabel.tap()
+            }
+            
+            test("AND She selects 'Receive a Code'") {
+                factorsPage.chooseButton.tap()
+            }
+        }
         
-        factorsPage.emailLabel.tap()
-        factorsPage.chooseButton.tap()
-        
-        let codePage = PasscodeFormPage(app: app)
-        XCTAssertTrue(codePage.passcodeLabel.waitForExistence(timeout: .regular))
-        XCTAssertTrue(codePage.passcodeField.exists)
-        
-        let emailCode = try receive(code: .email)
-        
-        codePage.passcodeField.tap()
-        codePage.passcodeField.typeText(emailCode)
-        
-        codePage.continueButton.tap()
+        try test("THEN the screen changes to receive an input for a code") {
+            let codePage = PasscodeFormPage(app: app)
+            XCTAssertTrue(codePage.passcodeLabel.waitForExistence(timeout: .regular))
+            XCTAssertTrue(codePage.passcodeField.exists)
+            
+            let emailCode = try scenario.receive(code: .email)
```

Comment:
You need to call the receive method on the test case class itself now, not the scenario object. That's because it needs to implement retry (the "Send again" button) because sometimes the first text message fails.